Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/ds/genericbatch/common/nhin/proxy/AbstractNhinX12BatchProxy.java

```diff
@@ -83,14 +78,20 @@ public COREEnvelopeBatchSubmissionResponse batchSubmitTransaction(COREEnvelopeBa
                 ServicePortDescriptor<GenericBatchTransactionPort> portDescriptor
                     = new COREGenericBatchServicePortDescriptor();
                 CONNECTClient<GenericBatchTransactionPort> client = getCONNECTClientSecured(portDescriptor, assertion,
-                    url, targetHCID, getServiceName());
+                    url, targetSystem, getServiceName());
                 client.enableMtom();
                 response = (COREEnvelopeBatchSubmissionResponse) client.invokePort(GenericBatchTransactionPort.class,
                     "batchSubmitTransaction", msg);
                 if (response != null && response.getPayload() != null) {
                     X12LargePayloadUtils.convertDataToFileLocationIfEnabled(response);
                 }
             } else {
+                String targetHCID = null;
```

Comment:
I see we have duplicate logic in here.  Consider to use HomeCommunityMap.getCommunityIdFromTargetCommunities()